### PR TITLE
ModuleUnloadLoad.sh code is now translated into python

### PR DESCRIPTION
**Pull request by Naresh-ibm**
ModuleUnloadLoad.sh was written in shell script and its been translated
into python code where we have existing library functions and easy to handle

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
this is in favour of #963

**Comment by Naresh-ibm** [comment]
[module_load_unload_only_io_job_log2_v2.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/2336218/module_load_unload_only_io_job_log2_v2.txt)

File: io/driver/module_unload_load.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
```
Comment (nasastry) [inline io/driver/module_unload_load.py]: 2018

File: io/driver/module_unload_load.py.data/README.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+This Program loads and unloads the kernel driver modules. The script takescare of unloading the dependant modules, if any, before unloading the core module. It runs for a total ofgiven number of iteration.
```
Comment (nasastry) [inline io/driver/module_unload_load.py.data/README.txt]: s/takescare/takes care/
s/ofgiven/of given/
s/iteration/iterations/

File: io/driver/module_unload_load.py.data/README.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+This script should be run as root.
+Inputs
+------
+ITERATIONS -    No of counts to unload and load the module. Defaults to 1.
```
Comment (nasastry) [inline io/driver/module_unload_load.py.data/README.txt]: s/No/Number/

File: io/driver/module_unload_load.py.data/README.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+MODULE -       Name of modules to unload/load.
+only_io-       if user wants to unload and load the specific IO module then this value should be True. False to unload and load all the module available in OS.
+fc-            if the module is of FC adapter then this value should be true which is used for flushing the multipath service without which we cannot unload the module. Flase if not FC adapter.:wq
```
Comment (nasastry) [inline io/driver/module_unload_load.py.data/README.txt]: s/:wq//

File: io/driver/module_unload_load.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                self.log.info("loading : %s " % mdl)
+                linux_modules.load_module(mdl)
+                time.sleep(3)
+
+    def test(self):
+        """
+        Begining the test here
+        """
+        pci_addrs = []
+        if self.only_io is True:
```
Comment (narasimhan-v) [inline io/driver/module_unload_load.py]: if `OnlyIO` is specified, all the pci devices' module should be unloaded and loaded

**Comment by PraveenPenguin** [comment]
@Naresh-ibm are you still perusing this PR

**Comment by PraveenPenguin** [comment]
closing this as it kind of enhancement to convert bash version to python @Naresh-ibm if you still interested to pursue this please raise new PR

**Pull request closed without merging**